src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java

```diff
@@ -1069,16 +1050,18 @@ public boolean isMute() {
   @Override
   public void setLocale(String code) {
     locale = new Locale(code);
+    loadLocalizations(locale.getLanguage());
+    log.info("{} new locale is {}", getName(), code);
   }
 
   @Override
   public String getLanguage() {
-    return locale.getLanguage();
+    return Runtime.getInstance().getLanguage();
   }
 
   @Override
   public Locale getLocale() {
-    return locale;
+    return Runtime.getInstance().getLocale();

```

Comment:
This should be the langauge of the current voice, not runtime.

Reply:
promoted normalized updated - no longer an issue